agent/agent.go

```diff
@@ -1260,182 +1267,183 @@ func (a *Agent) reloadWatches(cfg *config.RuntimeConfig) error {
 	return nil
 }
 
-// consulConfig is used to return a consul configuration
-func (a *Agent) consulConfig() (*consul.Config, error) {
+// newConsulConfig translates a RuntimeConfig into a consul.Config.
+// TODO: move this function to a different file, maybe config.go
+func newConsulConfig(config *config.RuntimeConfig, logger hclog.Logger) (*consul.Config, error) {
 	// Start with the provided config or default config
 	base := consul.DefaultConfig()
```

Comment:
This is a small nit on prior work, I don't like that this is called base. The name makes sense context of the beginning the fn, e.g. "this is the base that new params are built up on". But reading it at the end, it implies it was immutable, even though we modify it throughout the function. Returning the "base" after all of that work and then supplying a modified base (with the RuntimeConfig) to enterpriseConfig doesn't feel right.

This might be a good opportunity to rename `base` to `config` or `cfg` in this scope, and then rename the `RuntimeConfig` param to `newCfg` `runtimeCfg` - even `params` would work better. Then the `newConsulConfig` function takes the config defaults, overrides the defaults with params, and then returns *the* `config` at the end.

Reply:
Good call! I renamed `base` -> `cfg` and the RuntimeConfig arg to `runtimeCfg`.